services/wfm/src/main/java/org/openkilda/wfm/topology/flow/service/FlowService.java

```diff
@@ -0,0 +1,377 @@
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.flow.service;
+
+import static java.util.Objects.requireNonNull;
+
+import org.openkilda.model.Flow;
+import org.openkilda.model.FlowPair;
+import org.openkilda.model.FlowPath;
+import org.openkilda.model.FlowSegment;
+import org.openkilda.model.FlowStatus;
+import org.openkilda.model.Switch;
+import org.openkilda.pce.PathComputer;
+import org.openkilda.pce.PathComputerFactory;
+import org.openkilda.pce.PathComputerFactory.Strategy;
+import org.openkilda.pce.PathPair;
+import org.openkilda.pce.RecoverableException;
+import org.openkilda.pce.UnroutableFlowException;
+import org.openkilda.persistence.PersistenceManager;
+import org.openkilda.persistence.TransactionManager;
+import org.openkilda.persistence.repositories.FlowRepository;
+import org.openkilda.persistence.repositories.FlowSegmentRepository;
+import org.openkilda.persistence.repositories.IslRepository;
+import org.openkilda.persistence.repositories.RepositoryFactory;
+
+import com.google.common.collect.Lists;
+import lombok.extern.slf4j.Slf4j;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+@Slf4j
+public class FlowService {
+    private TransactionManager transactionManager;
+    private FlowRepository flowRepository;
+    private FlowSegmentRepository flowSegmentRepository;
+    private IslRepository islRepository;
+    private PathComputerFactory pathComputerFactory;
+    private FlowResourcesManager flowResourcesManager;
+
+    public FlowService(PersistenceManager persistenceManager, PathComputerFactory pathComputerFactory,
+                       FlowResourcesManager flowResourcesManager) {
+        transactionManager = persistenceManager.getTransactionManager();
+        RepositoryFactory repositoryFactory = persistenceManager.getRepositoryFactory();
+        flowRepository = repositoryFactory.createFlowRepository();
+        flowSegmentRepository = repositoryFactory.createFlowSegmentRepository();
+        islRepository = repositoryFactory.createIslRepository();
+        this.pathComputerFactory = requireNonNull(pathComputerFactory);
+        this.flowResourcesManager = requireNonNull(flowResourcesManager);
+    }
+
+    public boolean doesFlowExist(String flowId) {
+        return flowRepository.doesExistById(flowId);
+    }
+
+    public FlowPair getFlowPair(String flowId) {
+        Collection<FlowPair> pairs = flowRepository.findFlowPairById(flowId);
+        return pairs.isEmpty() ? null : pairs.iterator().next();
```

Comment:
I prefer exception or Optional<?> response. `null` is usually bad idea, especially for public method.